### PR TITLE
[test-strategy]Add lib-common specific rules

### PR DESCRIPTION
**Pull request by gibizer**
This is based on the comments from @stuggi in
https://github.com/openstack-k8s-operators/docs/pull/20

### CODE REVIEW

**Comment by gibizer** [comment]
I've added @fmount from storage and @slagle from edpm side as this affects the storage and ansible modules of lib-common too.

File: test_strategy.md [hunk 1]
```diff
@@ -40,3 +40,13 @@ coverage
 * Each PR changing implementation logic targeting the
 [network](https://github.com/openstack-k8s-operators/infra-operator/tree/main/controllers/network)
 controllers have to include relevant EnvTest coverage.
+
+### lib-common
```
Comment (fmount) [inline test_strategy.md:44]: Ack and +1 on my side. 
FWIW when we started the ceph module at the time (just helpers, not much used in the operators ecosystem) we added something like [1], which seems to fit the purpose.
I'm pretty sure we can improve the `storage/` module and add test cases for the existing functions, but in general for functional tests I would defer to the operators implementing the feature to cover the `scenario` and double check (via `envTest`) the expected results.
@gibizer how does it sounds?

[1] https://github.com/fmount/lib-common/blob/extra_volumes/modules/storage/ceph/funcs_test.go

**Comment by slagle** [comment]
+2 from me. We already have some initial unit test coverage in https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/ansible/inventory_test.go for the ansible related functionality.

File: test_strategy.md (hunk 1 above)
Comment (gibizer) [reply test_strategy.md:44]: Cool I will update the patch. I looked through the storage module and I don't see any k8s API calls in there so the existing unit test feel OK to me as is. We can discuss any incoming new features in that module separately but I will assume for now that unit test coverage will suffice.

**Comment by gibizer** [comment]
> +2 from me. We already have some initial unit test coverage in https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/ansible/inventory_test.go for the ansible related functionality.

Thanks. As far as I see the ansible module also does not reach out to k8s APIs so the existing unit test coverage is good as is and I will assume the same for the incoming new features for now as well.

File: test_strategy.md (hunk 1 above)
Comment (gibizer) [reply test_strategy.md:44]: Ohh I already have the unit / functional distinction in the current docs patch. So I think both the storage and ansible module is belonging to the unit tested category.

**Pull request merged**